libs/common/src/types/portal/activity.ts

```diff
@@ -0,0 +1,20 @@
+import { ActivityAuthor, ValuesOf } from '..';
+import { PORTAL_ACTIVITY_LABEL, PORTAL_ACTIVITY_TYPE } from '../../constants';
+
+export type ActivityType = ValuesOf<typeof PORTAL_ACTIVITY_TYPE>;
+export type ActivityLabel = ValuesOf<typeof PORTAL_ACTIVITY_LABEL>;
+
+export type Activity = {
+  type: ActivityType;
+  timestamp: number;
+  author: ActivityAuthor;
+  label: string;
+  activityType?: ActivityLabel;
```

Comment:
The data model/naming is confusing here.

```ts
type: ActivityType; // e.g "DEAL_CANCELLED", "FACILITY_STAGE", 

activityType: ActivityLabel; // e.g "Deal cancelled", "AIN submitted"
```

since we have two fields, `type` and `activityType`, i've opted to keep `type` aligned, and then `activityType` as `ActivityLabel`.

Not ideal but unless there are other ideas (and without data migration), I think this is clear as it could get - just means it's a little unclear in this one type/file.

Reply:
Could you add a comment here explaining this please :)

Reply:
Great shout @MarRobSoftwire have added.